src/odemis/gui/cont/export.py

```diff
@@ -168,15 +176,62 @@ def export_viewport(self, filepath, export_format, export_type):
             self._conf.last_export_path = os.path.dirname(filepath)
 
             exported_data = self.export(export_type, raw)
-            # record everything to a file
-            exporter.export(filepath, exported_data)
 
-            popup.show_message(self._main_frame,
-                               "Exported in %s" % (filepath,),
-                               timeout=3
-                               )
-
-            logging.info("Exported a %s view into file '%s'.", export_type, filepath)
+            # batch export
+            # TODO: for now we do not create a new folder where all files are saved
+
+            if isinstance(exported_data, dict):
+                # get the file names
+                filename_dict = {}
+                for key in exported_data.keys():
+                    base = os.path.basename(filepath)
+                    try:  # .png
+                        base_name = os.path.splitext(os.path.splitext(base)[0])[0]
+                        file_extension = os.path.splitext(os.path.splitext(base)[0])[1] + os.path.splitext(base)[1]
+                    except AttributeError:  # .ome.tiff
+                        base_name = os.path.splitext(base)[0]
+                        file_extension = os.path.splitext(base)[1]
+                    dir_name = os.path.dirname(filepath)
+                    # use the filename defined by the user and add the MD_POL_MODE to the filename
+                    filename = os.path.join(dir_name, base_name + "_" + key + file_extension)
+
+                    # detect we'd overwrite an existing file => show our own warning
+                    if os.path.exists(filename):
+                        dlg = wx.MessageDialog(self._main_frame,
+                                               "A file named \"%s\" already exists.\n"
+                                               "Do you want to replace it?" % (filename,),
+                                               "File already exists",
+                                               wx.YES_NO | wx.NO_DEFAULT | wx.ICON_QUESTION)
+                        ret = dlg.ShowModal()
+                        dlg.Destroy()
+                        if ret == wx.ID_NO:
+                            return
+
+                    filename_dict[key] = filename
+
+                # record everything to a file
+                for key in exported_data.keys():
+                    exporter.export(filename_dict[key], exported_data[key])
+
+                # TODO need to be adapted for batch export as now redundant
```

Comment:
What does this TODO mean? Is that still true? What is redundant?

Reply:
Yes this TODO is still True, as the popup.show_message is still duplicated for single file and batch export. See next lines...